build/common.sh

```diff
@@ -950,6 +950,20 @@ function kube::release::package_kube_manifests_tarball() {
   mkdir -p "${dst_dir}/dns"
   tar c -C "${KUBE_ROOT}/cluster/saltbase/salt/kube-dns" ${objects} | tar x -C "${dst_dir}/dns"
 
+  # We leave the `{{ pillar['federations_domain_map'] }}` parameter as is, if
+  # the right federation environment variables isn't set. This is to allow
+  # users to provide these pillar values using the regular salt's mechanisms
+  # during cluster bootstrap.
+  if [[ "${FEDERATION:-}" == "true" ]]; then
+    FEDERATIONS_DOMAIN_MAP="${FEDERATIONS_DOMAIN_MAP:-}"
+    if [[ -z "${FEDERATIONS_DOMAIN_MAP}" && -n "${FEDERATION_NAME:-}" && -n "${DNS_ZONE_NAME:-}" ]]; then
```

Comment:
Maybe its because I dont know enough about salt: but why would someone want to override it using salt rather than doing it using env var? The advantage of doing it using env var is that it sets the right values in both kube-dns and the controller manager. (salt will only set it in kube-dns).
The problem I see with current setup is that it can happen that I dont set the values anywhere (neither in env var nor in salt), my cluster bring up will work fine. Just that my DNS resolution for federation queries will not work.
If we decide not to set default values for FEDERATION_NAME and DNS_ZONE_NAME, then cluster bring up script should throw an error if FEDERATION=true and FEDERATION_NAME or DNS_ZONE_NAME are not specified.